### DISCUSSION TITLE
not importing properly

### DISCUSSION THREAD

**Discussion by AcademicRoutine168** [noDRM/DeDRM_tools#556] [Q&A]
I installed most up to date windows kindle app and dedrm v10.0.9. i cannot convert kindle books. i extracted my key from kindle app. 

Log file follows:

`ApplicationPaletteChange event ignored
Using calibre Qt style: True
ApplicationPaletteChange event received
calibre Debug log
calibre 7.11  embedded-python: True
Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19045')
Python 3.11.5
Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Interface language: None
EXE path: C:\Program Files\Calibre2\calibre-debug.exe
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (2, 11, 0) && DeDRM (10, 0, 9) && Package KFX (from KFX Input) (2, 11, 0) && Goodreads (1, 8, 1) && KFX metadata reader (from KFX Input) (2, 11, 0) && From KFX (2, 11, 0) && KFX Input (2, 11, 0) && Set KFX metadata (from KFX Output) (2, 8, 0) && KFX Output (2, 8, 0)
calibre 7.11  embedded-python: True
Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19045')
Python 3.11.5
Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Interface language: None
EXE path: C:\Program Files\Calibre2\calibre-debug.exe
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (2, 11, 0) && DeDRM (10, 0, 9) && Package KFX (from KFX Input) (2, 11, 0) && Goodreads (1, 8, 1) && KFX metadata reader (from KFX Input) (2, 11, 0) && From KFX (2, 11, 0) && KFX Input (2, 11, 0) && Set KFX metadata (from KFX Output) (2, 8, 0) && KFX Output (2, 8, 0)
QPA platform: windows
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 69.86819484240687 x 69.80152671755725
[0.00] Starting up...
[0.00] Showing splash screen...
[0.14] splash screen shown
[0.14] Initializing db...
[0.81] db initialized
[0.81] Constructing main UI...
[1.67] GUI main window shown
[1.73] main UI initialized...
[1.73] Hiding splash screen
[15.50] splash screen hidden
[15.50] Started up in 15.50 seconds with 463 books
stdout+stderr from file dialog helper: [b'', b'']
piped data from file dialog helper: [b'\xcfz\x7f\xce\x92\x83Xy\x0c\x91C\x7f-\x13L/\xe9\x81\xc1\xb3\xcd\xf6m^\xa7\xe7\x9b\x95\xf2\xd2\x89\\', b'G:\\backup\\documents\\My Kindle Content\\B0C2J6Z786_EBOK\\CR!00TYWZHZ9D2H5BZT57NZV586G17M.azw.md']
calibre 7.11  embedded-python: True
Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19045')
Python 3.11.5
Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Interface language: None
EXE path: C:\Program Files\Calibre2\calibre-debug.exe
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (2, 11, 0) && DeDRM (10, 0, 9) && Package KFX (from KFX Input) (2, 11, 0) && Goodreads (1, 8, 1) && KFX metadata reader (from KFX Input) (2, 11, 0) && From KFX (2, 11, 0) && KFX Input (2, 11, 0) && Set KFX metadata (from KFX Output) (2, 8, 0) && KFX Output (2, 8, 0)
Traceback (most recent call last):
  File "calibre\gui2\actions\view.py", line 324, in view_triggered
  File "calibre\gui2\actions\view.py", line 401, in _view_books
  File "calibre\gui2\actions\view.py", line 353, in _view_calibre_books
  File "calibre\gui2\actions\view.py", line 148, in view_format_by_id
  File "calibre\gui2\actions\view.py", line 206, in _view_file
  File "calibre\gui2\actions\view.py", line 198, in _launch_viewer
  File "calibre\gui2\__init__.py", line 1570, in open_local_file
OSError: [WinError -2147221003] Application not found: 'G:\\ebooks\\Unknown\\CR!00TYWZHZ9D2H5BZT57NZV586G17M.azw (3930)\\CR!00TYWZHZ9D2H5BZT57NZV586G17M - Unknown.md'

Convert book 1 of 1 (CR!00TYWZHZ9D2H5BZT57NZV586G17M.azw)

Conversion options changed from defaults:

  verbose: 2

  read_metadata_from_opf: 'C:\\Users\\shawn\\AppData\\Local\\Temp\\calibre_p4rsbw29\\cvs4e3xa.opf'

  output_profile: 'kindle_fire'

Resolved conversion options

calibre version: 7.11.0

{'asciiize': False,

 'author_sort': None,

 'authors': None,

 'base_font_size': 0.0,

 'book_producer': None,

 'change_justification': 'original',

 'chapter': "//*[((name()='h1' or name()='h2') and re:test(., "

            "'\\s*((chapter|book|section|part)\\s+)|((prolog|prologue|epilogue)(\\s+|$))', "

            "'i')) or @class = 'chapter']",

 'chapter_mark': 'pagebreak',

 'comments': None,

 'cover': None,

 'debug_pipeline': None,

 'dehyphenate': True,

 'delete_blank_paragraphs': True,

 'disable_font_rescaling': False,

 'dont_compress': False,

 'duplicate_links_in_toc': False,

 'embed_all_fonts': False,

 'embed_font_family': None,

 'enable_heuristics': False,

 'expand_css': False,

 'extra_css': None,

 'extract_to': None,

 'filter_css': '',

 'fix_indents': True,

 'font_size_mapping': None,

 'format_scene_breaks': True,

 'formatting_type': 'auto',

 'html_unwrap_factor': 0.4,

 'input_encoding': None,

 'input_profile': <calibre.customize.profiles.InputProfile object at 0x00000243EA17A8D0>,

 'insert_blank_line': False,

 'insert_blank_line_size': 0.5,

 'insert_metadata': False,

 'isbn': None,

 'italicize_common_cases': True,

 'keep_ligatures': False,

 'language': None,

 'level1_toc': None,

 'level2_toc': None,

 'level3_toc': None,

 'line_height': 0.0,

 'linearize_tables': False,

 'margin_bottom': 5.0,

 'margin_left': 5.0,

 'margin_right': 5.0,

 'margin_top': 5.0,

 'markdown_extensions': 'footnotes, tables, toc',

 'markup_chapter_headings': True,

 'max_toc_links': 50,

 'minimum_line_height': 120.0,

 'mobi_file_type': 'old',

 'mobi_ignore_margins': False,

 'mobi_keep_original_images': False,

 'mobi_toc_at_start': False,

 'no_chapters_in_toc': False,

 'no_inline_navbars': True,

 'no_inline_toc': False,

 'output_profile': <calibre.customize.profiles.KindleFireOutput object at 0x00000243EA053D90>,

 'page_breaks_before': "//*[name()='h1' or name()='h2']",

 'paragraph_type': 'auto',

 'personal_doc': '[PDOC]',

 'prefer_author_sort': False,

 'prefer_metadata_cover': False,

 'preserve_spaces': False,

 'pretty_print': False,

 'pubdate': None,

 'publisher': None,

 'rating': None,

 'read_metadata_from_opf': 'C:\\Users\\shawn\\AppData\\Local\\Temp\\calibre_p4rsbw29\\cvs4e3xa.opf',

 'remove_fake_margins': True,

 'remove_first_image': False,

 'remove_paragraph_spacing': False,

 'remove_paragraph_spacing_indent_size': 1.5,

 'renumber_headings': True,

 'replace_scene_breaks': '',

 'search_replace': '[]',

 'series': None,

 'series_index': None,

 'share_not_sync': False,

 'smarten_punctuation': False,

 'sr1_replace': None,

 'sr1_search': None,

 'sr2_replace': None,

 'sr2_search': None,

 'sr3_replace': None,

 'sr3_search': None,

 'start_reading_at': None,

 'subset_embedded_fonts': False,

 'tags': None,

 'timestamp': None,

 'title': None,

 'title_sort': None,

 'toc_filter': None,

 'toc_threshold': 6,

 'toc_title': None,

 'transform_css_rules': '[]',

 'transform_html_rules': '[]',

 'txt_in_remove_indents': False,

 'unsmarten_punctuation': False,

 'unwrap_lines': True,

 'use_auto_toc': False,

 'verbose': 2}

InputFormatPlugin: TXT Input running

on C:\Users\shawn\AppData\Local\Temp\calibre_p4rsbw29\752wgmha.md

Reading text from file...

File extension indicates particular formatting. Forcing formatting type to: markdown

Detected input encoding as windows-1252 with a confidence of 100%

Running text through markdown conversion...

Language not specified

Creator not specified

Building file list...

Traceback (most recent call last):

  File "runpy.py", line 198, in _run_module_as_main

  File "runpy.py", line 88, in _run_code

  File "site.py", line 83, in <module>

  File "site.py", line 78, in main

  File "site.py", line 50, in run_entry_point

  File "calibre\utils\ipc\worker.py", line 215, in main

  File "calibre\gui2\convert\gui_conversion.py", line 38, in gui_convert_override

  File "calibre\gui2\convert\gui_conversion.py", line 25, in gui_convert

  File "calibre\ebooks\conversion\plumber.py", line 1122, in run

  File "calibre\customize\conversion.py", line 245, in __call__

  File "calibre\ebooks\conversion\plugins\txt_input.py", line 325, in convert

  File "calibre\ebooks\conversion\plugins\html_input.py", line 109, in convert

  File "calibre\ebooks\conversion\plugins\html_input.py", line 171, in create_oebbook

  File "calibre\ebooks\html\input.py", line 296, in get_filelist

  File "calibre\ebooks\html\input.py", line 271, in traverse

  File "calibre\ebooks\html\input.py", line 111, in __init__

ValueError: embedded null character

IgnoreFile("Could not read from file: C:\\Users\\shawn\\AppData\\Local\\Temp\\calibre_p4rsbw29\\fqki9ng9_plumber\\HÓPš\x14ö¢”I�    ˜� with error: [Errno 22] Invalid argument: 'C:\\\\Users\\\\shawn\\\\AppData\\\\Local\\\\Temp\\\\calibre_p4rsbw29\\\\fqki9ng9_plumber\\\\HÓPš\\x14ö¢”I�    ˜�'")

`

**Comment by ElleKayEm** [comment]
It looks like you may be importing the wrong file.  You want the one with just the azw extension and this appears to be azw.md.  Do you have extensions hidden in Windows Explorer?

**Reply by AcademicRoutine168** [reply]
you are correct. i was trying to input the wrong file in the folder. this issue is resolved. i'm an idiot :)

**Comment by AcademicRoutine168** [comment]
wrong file selected in folder, no further issues
